controllers/product.go

```diff
@@ -0,0 +1,282 @@
+		return
+	}
+	productt.IsActive = *req.IsActive
+
+	_, err = productt.Update()
+	if err != nil {
+		_ = c.Error(err)
+		c.JSON(http.StatusOK, respond.UpdatedFail())
+		return
+	}
+	c.JSON(http.StatusOK, respond.Success(productt.IsActive, "update successfully"))
+}
+
+// Function Delete
+func (productClt ProductController) Delete(c *gin.Context) {
+	productModel := new(models.Products)
+	var reqUri request.DeleteUri
+	err := c.ShouldBindUri(&reqUri)
+	if err != nil {
+		_ = c.Error(err)
+		c.JSON(http.StatusBadRequest, respond.MissingParams())
+		return
+	}
+
+	condition := bson.M{"uuid": reqUri.Uuid}
+	productt, err := productModel.FindOne(condition)
+	if err != nil {
+		fmt.Println(err.Error())
+		c.JSON(http.StatusOK, respond.ErrorCommon("Product no found!"))
+		return
+	}
+
+	productt.IsDelete = constant.DELETE
+
+	_, err = productt.Update()
+	if err != nil {
+		fmt.Println(err.Error())
+		c.JSON(http.StatusOK, respond.UpdatedFail())
+		return
+	}
+	c.JSON(http.StatusOK, respond.Success(productt.Uuid, "Delete successfully"))
+}
+
+// Function Create
+func (productClt ProductController) Create(c *gin.Context) {
+	var req request.GetInsertRequest
+	err := c.ShouldBindWith(&req, binding.Query)
+	if err != nil {
+		_ = c.Error(err)
+		c.JSON(http.StatusBadRequest, respond.MissingParams())
+		return
+	}
+	err = c.ShouldBindJSON(&req)
+	if err != nil {
+		_ = c.Error(err)
+		c.JSON(http.StatusBadRequest, respond.MissingParams())
+		return
+	}
+	productData := models.Products{}
+	productData.Uuid = util.GenerateUUID()
+	productData.Name = req.Name
+	productData.Image = req.Image
+	productData.IsActive = req.IsActive
+	productData.Price = req.Price
+	productData.Quantity = req.Quantity
+	productData.Sequence = req.Sequence
+	productData.StartDate = req.StartDate
+	productData.EndDate = req.EndDate
+	productData.Gallery = req.Gallery
+	productData.Description = req.Description
+	_, err = productData.Insert()
+	if err != nil {
+		fmt.Println(err.Error())
+		c.JSON(http.StatusOK, respond.UpdatedFail())
+		return
+	}
+	c.JSON(http.StatusOK, respond.Success(productData.Uuid, "Successfully"))
+}
+
+// Function Update
+func (productClt ProductController) Update(c *gin.Context) {
+	productModel := new(models.Products)
+	var reqUri request.UpdateUriDetail
+	err := c.ShouldBindUri(&reqUri)
+	if err != nil {
+		_ = c.Error(err)
+		c.JSON(http.StatusBadRequest, respond.MissingParams())
+		return
+	}
+	var req request.UpdateDetailRequest
+	err = c.ShouldBindJSON(&req)
+	if err != nil {
+		_ = c.Error(err)
+		c.JSON(http.StatusBadRequest, respond.MissingParams())
+		return
+	}
+
+	condition := bson.M{"uuid": reqUri.Uuid}
+	productt, err := productModel.FindOne(condition)
+	if err != nil {
+		fmt.Println(err.Error())
+		c.JSON(http.StatusOK, respond.ErrorCommon("Product no found!"))
+		return
+	}
+	if req.IsActive != nil {
+		productt.IsActive = *req.IsActive
+	}
+	if req.Name != "" {
+		productt.Name = req.Name
+	}
+	if req.Image != "" {
```

Comment:
Field Image, Description, Price, Sequence, Quantity can be optional, can set with default value.
Missing Gallery field